### ISSUE TITLE
[Enhancement] Allow autocomplete (from the browser history) in the search box

### ISSUE THREAD

**Issue by raphj** [iv-org/invidious#3704]
**Is your enhancement request related to a problem? Please describe.**

I'd like to be able to use the browser history for the search bar. Especially on touchscreen where it is more difficult to type. It's annoying to have to type things completely again and again.

**Describe the solution you'd like**

This is caused by the `autocompete="off"` attribute in `views/components/search_box.ecr`.

```
                <input type="search" id="searchbox" autocomplete="off" autocorrect="off"
                autocapitalize="none" spellcheck="false" <% if autofocus %>autofocus<% end %>
                name="q" placeholder="<%= translate(locale, "search") %>"
                title="<%= translate(locale, "search") %>"
                value="<%= env.get?("search").try {|x| HTML.escape(x.as(String)) } %>">
```

I'd like that this attribute to be dropped.

@SamantazFox it seems this comes from your commit 68cbc1181 to fix #1107 (nice commit documentation! thanks for this). `spellcheck="false"` or `autocorrect="off"` seem necessary to avoid annoying autocorrection, do you remember why you introduced `autocomplete="off"` as well?

**Describe alternatives you've considered**

- a setting to disable `autocomplete=off`
- a local commit that disables it, since I'm currently building invidious myself (but that might not always be the case in the future)
- a user script that patches the search box
- an Invidious feature that remembers, per user, their search history but doesn't seem ideal to store such sensitive data server side.

**Additional context**

I can provide a PR if it helps.

**Comment by unixfox** [comment]
Do note that there is no autocomplete system in the backend, so even if you were to enable it, nothing would be able to provide the autocompletion to the browser.

For the autocompletion to work, we need someone to implement this feature on the backend side.

**Comment by raphj** [comment]
No no, when I remove the attribute, the browser suggests stuff I already typed there :-)

**Comment by unixfox** [comment]
Then we are not talking about the same autocompletion, I'm talking about the autocompletion that YouTube servers would give. The same behavior that you would have when typing in the search bar of Google for example.

**Comment by raphj** [comment]
I'm referring to the native feature of browsers that is disabled by `autocomplete=off`

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete

Google would have it, but they disable it to provide their own (that indeed requires backend support).

**Comment by raphj** [comment]
There's a risk browsers auto-fill the search bar with some previous search, I think it can be avoided by also setting `value=""`

**Comment by raphj** [comment]
Note that autocomplete="off" might not work on Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=587466

I'm using Firefox, which honors this value.

**Comment by unixfox** [comment]
Please use the edit feature of github instead of posting a new message. If possible group all your 3 previous messages.
No need to reply, just add the :eye: reaction.

**Comment by SamantazFox** [comment]
> @SamantazFox it seems this comes from your commit [68cbc11](https://github.com/iv-org/invidious/commit/68cbc11810fd746a776f6b8fefd61cec89ac9b55) to fix #1107 (nice commit documentation! thanks for this). `spellcheck="false"` or `autocorrect="off"` seem necessary to avoid annoying autocorrection, do you remember why you introduced `autocomplete="off"` as well?

I don't recall why I did that, sorry. Maybe it was requested in a github issue?

**Comment by SamantazFox** [comment]
Closed by https://github.com/iv-org/invidious/pull/3724
